test/StandardBasis.jl

```diff
@@ -984,6 +984,10 @@ end
             b = fromroots(r)
             (b ≈ a) & isreal(coeffs(b))    # the coeff should be real
         end
+        p = Polynomial([1; zeros(99); -1])
+        if P !== FactoredPolynomial
+            @test fromroots(P, roots(p)) * p[end] ≈ p
```

Comment:
For `FactoredPolynomial`, `fromroots` is trivially correct; the actual conversion happens when promoting to a common type with `p`. That goes through `evalpoly` which uses an arbitrary ordering of the roots. I wonder whether it would generally be preferable to use the Leja ordering in `evalpoly` here, in which case it would probably be worth it to replace the `Dict` in `FactoredPolynomial` with an `OrderedDict` and do the Leja ordering upon construction. No clue whether that's a good idea.

Reply:
Well that's a question for another day, but it seems like a reasonable thing to do. The extra package has negligible load time.